### PR TITLE
[Storybook] fix warning and improve perf

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

File: packages/storybook/.storybook/main.js [hunk 1]
```diff
@@ -24,10 +24,11 @@ module.exports = {
     '@storybook/addon-a11y/register',
   ],
   typescript: {
-    check: __DEV__, // Netlify is breaking the deploy with this settings on. So deactivate on release
+    check:  __DEV__, // Netlify is breaking the deploy with this settings on. So deactivate on release
```
Comment (oliviertassinari) [inline packages/storybook/.storybook/main.js:27]: AFAIK, this should be formated by prettier but isn't.

@eps1lon By default, we ignore dot files/folders, shouldn't set [`dot: true`](https://github.com/isaacs/node-glob#dots) in the [glob options](https://github.com/mui-org/material-ui/blob/cb6b69baa828623393b1ed0ade8efde309ac06b0/scripts/prettier.js#L33)?

File: packages/storybook/.storybook/main.js [hunk 2]
```diff
@@ -24,10 +24,11 @@ module.exports = {
...
+    reactDocgen: false
   },
   webpackFinal: async config => {
-    config.devtool = __DEV__ ? 'inline-source-map' : undefined;
+    // config.devtool = __DEV__ ? 'eval-source-map' : undefined;
```
Comment (oliviertassinari) [inline packages/storybook/.storybook/main.js:31]:
```suggestion
```

**Review by oliviertassinari — COMMENTED** [review]
Without a description of the pull request, I can't know what I should review.

File: packages/storybook/.storybook/main.js (hunk 1 above)
Comment (eps1lon) [reply packages/storybook/.storybook/main.js:27]: For prettier that makes sense, yes.

Comment (oliviertassinari) [reply packages/storybook/.storybook/main.js:27]: Ok, I will add it in the next batch. It should be a one-line change. Then, once committed we can benefit from it on material-ui-x.

File: packages/storybook/src/stories/playground/real-data-demo.stories.tsx [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import * as React from 'react';
-import { withA11y } from '@storybook/addon-a11y';
```
Comment (oliviertassinari) [inline packages/storybook/src/stories/playground/real-data-demo.stories.tsx:2]: Does it mean that we should remove `@storybook/addon-a11y` from the list of dependencies?

File: packages/storybook/.storybook/main.js (hunk 1 above)
Comment (eps1lon) [reply packages/storybook/.storybook/main.js:27]: > It should be a one-line change.

Those are usually the kind of changes the lead to days of work. At least I thought so in https://github.com/mui-org/material-ui/pull/22814: Just bump the browser version and done :laughing:

File: packages/storybook/src/stories/playground/real-data-demo.stories.tsx (hunk 3 above)
Comment (dtassone) [reply packages/storybook/src/stories/playground/real-data-demo.stories.tsx:2]: Nope, we just don't need to declare it here

**Comment by dtassone** [comment]
> Without a description of the pull request, I can't know what I should review.

As mentioned, this PR fix all the warnings and improve the SB build time

**Comment by dtassone** [comment]
70+ sec down to 30 sec

**Comment by oliviertassinari** [comment]
I have recently found https://vitro.now.sh/. I wonder if we shouldn't consider it as an alternative to Storybook. Vitro comes with Next.js built-in. Maybe it could nail the performance issue of Storybook. I mean: check this benchmark: https://www.webpagetest.org/video/compare.php?tests=201007_DiKP_23fb114e7e169261238c212e78d5b336,201007_DiBM_e569507ebd60a8b9e5075b8b479b3a07

Capture d’écran 2020-10-07 à 15 25 34

Something could be improved

**Comment by dtassone** [comment]
> I have recently found https://vitro.now.sh/. I wonder if we shouldn't consider it as an alternative to Storybook. Vitro comes with Next.js built-in. Maybe it could nail the performance issue of Storybook. I mean: check this benchmark: https://www.webpagetest.org/video/compare.php?tests=201007_DiKP_23fb114e7e169261238c212e78d5b336,201007_DiBM_e569507ebd60a8b9e5075b8b479b3a07
> 
> Capture d’écran 2020-10-07 à 15 25 34
> 
> Something could be improved

Still very early stage
No Addon like Knobs and all the others in the pipeline

**Comment by oliviertassinari** [comment]
Yeah, it's probably too early.

**Pull request merged**